### PR TITLE
[CPDLP-2026] Update API V3 draft spec

### PR DESCRIPTION
**Pull request by cwrw**
### Context
We are ready to release the new API V3 draft spec

- Ticket: https://dfedigital.atlassian.net/jira/software/projects/CPDLP/boards/87?selectedIssue=CPDLP-2026

### Changes proposed in this pull request

- Add new endpoints as per the document in the ticket, as well as update existing ones.
- Add a release note (set for next week)

### Guidance to review
This is quite hard to follow, I suggest going through this commit by commit as I've grouped the same endpoints in each commit. Also to use the review app to help spot any issues

### CODE REVIEW

File: swagger/v3/api_spec.json [hunk 1]
```diff
@@ -1762,6 +2154,130 @@
...
+                }
+              }
+            }
+          }
+        }
+      }
+    },
     "/api/v3/statements": {
       "get": {
         "summary": "Retrieve financial statements",
```
Comment (twh32) [inline swagger/v3/api_spec.json:2283]: Note, this endpoint is new.

File: swagger/v3/api_spec.json [hunk 2]
```diff
@@ -1504,12 +1754,12 @@
         }
       }
     },
-    "/api/v3/participants/ecf/{id}/defer": {
-      "put": {
-        "summary": "Notify that an ECF participant is taking a break from their course",
```
Comment (twh32) [inline swagger/v3/api_spec.json:1509]: Note this endpoint includes an updated specification.

File: spec/docs/v3/ecf_schools_spec.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                schema: {
+                  "$ref": "#/components/schemas/ECFSchoolsSort",
+                },
+                style: :form,
+                explode: false,
+                required: false,
+                description: "Sort schools being returned.",
+                example: "sort=-updated_at"
+
+      response "200", "A list of ECF schools" do
```
Comment (twh32) [inline spec/docs/v3/ecf_schools_spec.rb:39]: Perhaps "A list of schools for the given cohort". (Just thinking that cohort is a required param/filter in the URL)

File: swagger/v3/component_schemas/ECFEnrolment.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    type: string
+    example: active
+    enum:
+      - active
+      - deferred
+      - withdrawn
+  participant_status:
+    description: Replaces the old status field. Indicates if a SIT has advised DfE of a transfer or a withdrawal of the participant from the school
+    type: string
+    enum:
```
Comment (twh32) [inline swagger/v3/component_schemas/ECFEnrolment.yml:61]: Do we want to add 'left' as a value here. It applies once a leaving_date is passed.

File: swagger/v3/component_schemas/ECFParticipantTransfers.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+description: "The details of an ECF Participant enrolment"
+type: object
+required:
+  - training_record_id
+  - transfer_type
+  - status
+  - created_at
+properties:
+  training_record_id:
+    description: "The unique identifier of the participant training record"
```
Comment (twh32) [inline swagger/v3/component_schemas/ECFParticipantTransfers.yml:10]: Do we want to add a line. "Should the DfE dedupe a participant, this value will not change."

File: swagger/v3/api_spec.json [hunk 6]
```diff
@@ -3021,15 +3794,219 @@
...
+          "in_partnership": {
+            "description": "The cohort for which you are reporting the partnership",
+            "type": "boolean",
+            "example": true
+          },
+          "induction_programme_choice": {
+            "description": "The induction programme the school offers",
+            "type": "string",
+            "example": "not_yet_known",
+            "enum": [
```
Comment (twh32) [inline swagger/v3/api_spec.json:3919]: is "funded_induction_programme" not a value here? (Would cover schools doing FIP with lead providers)

File: swagger/v3/api_spec.json [hunk 7]
```diff
@@ -3021,15 +3794,219 @@
...
+            "type": "string",
+            "example": "123456"
+          },
+          "cohort": {
+            "description": "The cohort the school is in",
+            "type": "string",
+            "example": 2021
+          },
+          "in_partnership": {
+            "description": "The cohort for which you are reporting the partnership",
```
Comment (twh32) [inline swagger/v3/api_spec.json:3911]: "Whether or not the school already has an active partnership, if it is doing a funded_induction_programme"

File: swagger/v3/component_schemas/ECFParticipantAttributes.yml [hunk 8]
```diff
@@ -1,130 +1,25 @@
...
-    example: active
-    enum:
-      - active
-      - deferred
-      - withdrawn
-  schedule_identifier:
-    description: "The schedule of the ECF participant"
-    type: string
-    example: ecf-standard-january
-    enum:
```
Comment (twh32) [inline swagger/v3/component_schemas/ECFParticipantAttributes.yml:78]: I think this is where it says "possible values". Would be tempted to remove as the range of possible values is much longer i.e. all the non-standard schedules. 

Perhaps instead we say: "See ECF change schedule guidance: https://manage-training-for-early-career-teachers.education.gov.uk/api-reference/ecf-usage.html#notifying-of-schedule-change"

File: swagger/v3/component_schemas/ECFEnrolment.yml [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    example: true
+  sparsity_uplift:
+    description: "Indicates whether this participant qualifies for an uplift payment due to sparsity"
+    type: boolean
+    example: true
+  schedule_identifier:
+    description: "The schedule of the ECF participant"
+    type: string
+    example: ecf-standard-january
+    enum:
```
Comment (twh32) [inline swagger/v3/component_schemas/ECFEnrolment.yml:88]: I think this is where it says "possible values". Would be tempted to remove as the range of possible values is much longer i.e. all the non-standard schedules. 

Perhaps instead we say: "See ECF change schedule guidance: https://manage-training-for-early-career-teachers.education.gov.uk/api-reference/ecf-usage.html#notifying-of-schedule-change"

**Review by mooktakim — APPROVED** [review]

**Review by mooktakim — APPROVED** [review]

**Review by leandroalemao — APPROVED** [review]
LGTM 👍  tksss
